core/src/main/java/org/elasticsearch/index/translog/TruncateTranslogCommand.java

```diff
@@ -0,0 +1,217 @@
+import org.elasticsearch.cli.SettingCommand;
+import org.elasticsearch.cli.Terminal;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.io.PathUtils;
+import org.elasticsearch.index.IndexNotFoundException;
+import org.elasticsearch.index.translog.Checkpoint;
+
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.nio.file.DirectoryStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardCopyOption;
+import java.nio.file.StandardOpenOption;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public class TruncateTranslogCommand extends SettingCommand {
+
+    /** A prefix of the bytes for an empty translog, the translog UUID is appended afterward */
+    private static byte[] EMPTY_TRANSLOG_PREFIX = new byte[] {
+        // lucene magic number (int)
+        0x3f, (byte)0xd7, 0x6c, 0x17,
+        // translog + lucene header
+        0x08, 0x74, 0x72, 0x61, 0x6e, 0x73, 0x6c, 0x6f, 0x67, 0x00, 0x00, 0x00, 0x02,
+        // length of translog UUID (int)
+        0x00, 0x00, 0x00, 0x16 };
+
+    private final OptionSpec<String> translogFolder;
+    private final OptionSpec<Void> batchMode;
+
+    public TruncateTranslogCommand() {
+        super("Truncates a translog to create a new, empty translog");
+        this.translogFolder = parser.acceptsAll(Arrays.asList("d", "dir"),
+                "Translog Directory location on disk")
+                .withRequiredArg()
+                .required();
+        this.batchMode = parser.acceptsAll(Arrays.asList("b", "batch"),
+                "Enable batch mode explicitly, automatic confirmation of warnings");
+    }
+
+    // Visible for testing
+    public OptionParser getParser() {
+        return this.parser;
+    }
+
+    @Override
+    protected void printAdditionalHelp(Terminal terminal) {
+        terminal.println("This tool truncates the translog and translog");
+        terminal.println("checkpoint files to create a new translog");
+    }
+
+    @Override
+    protected void execute(Terminal terminal, OptionSet options, Map<String, String> settings) throws Exception {
+        boolean batch = options.has(batchMode);
+
+        Path translogPath = PathUtils.get(translogFolder.value(options), "", "");
+        Path idxLocation = translogPath.getParent().resolve("index");
+
+        if (Files.exists(translogPath) == false || Files.isDirectory(translogPath) == false) {
+            throw new ElasticsearchException("translog directory [" + translogPath + "], must exist and be a directory");
+        }
+
+        if (Files.exists(idxLocation) == false || Files.isDirectory(idxLocation) == false) {
+            throw new ElasticsearchException("unable to find a shard at [" + idxLocation + "], which must exist and be a directory");
+        }
+
+        // Hold the lock open for the duration of the tool running
```

Comment:
Oh okay! I thought the directory automatically acquired the lock. I'll add this and add a test that checks it as you suggested.